### ISSUE TITLE
Trailing whitespace added when using --wrap

### ISSUE THREAD

**Issue by MrMino** [KyleKing/mdformat-mkdocs#20]
When using `mdformat` with `mdformat-mkdocs` a trailing whitespace is added to the file, triggering trailing whitespace check, and making it impossible to pass `pre-commit`.

This only happens when the `--wrap` option is specified.

Markdown:

```
- first line first line first line first line first line first line first line
    whitespace 
- second line
```

_(note the trailing whitespace on the second line:  `whitespace `)_

pre-commit config:

```
repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.5.0
    hooks:
    -   id: trailing-whitespace
-   repo: https://github.com/executablebooks/mdformat
    rev: 0.7.17
    hooks:
    -   id: mdformat
        args: ["--wrap", "79"]
        additional_dependencies:
          - mdformat-mkdocs>=2.0.0
```

Running `pre-commit run --all` with the markdown file staged makes it continuously report errors in both `trailing-whitespace` and `mdformat`, but no changes are made to the file in the end; the trailing whitespace hook removes the whitespace, mdformat adds it back.

**Comment by KyleKing** [comment]
Thanks for trying out `mdformat-mkdocs`!

I've added a test case, but I'm not able to replicate the conflict (`mdformat-mkdocs` fixes the indent on the second line and removes the trailing space): https://github.com/KyleKing/mdformat-mkdocs/commit/acfbcf5009be72004f2093c9de5fab76d04ea273

I think what might be happening is that both pre-commit hooks are trying to modify the same line, so pre-commit resets the line and reporting errors. If you try running just `mdformat` (`pre-commit run --all-files mdformat`), the line should be properly formatted. Afterwards, you should be unblocked to run pre-commit and both steps should complete without error

**Comment by KyleKing** [comment]
I ran a quick check to confirm and you're right that this is weird. `mdformat-mkdocs` is restoring the single space 😕

```sh
mkcd mdformat-mkdocs-ticket-20
pbpaste > .pre-commit-config.yaml
pbpaste > tmp.md
git init
git stage --all

pre-commit run --all-files
# Both error

pre-commit run --all-files mdformat
# No change

pre-commit run --all-files trailing-whistepace
# Removes space

git stage --all
pre-commit run --all-files mdformat
# Restores the space?
```

**Comment by KyleKing** [comment]
I'll try to release an update over the next week with work and vacation. My initial guess is that the filler characters used to make wrapping possible have an edge case where multiple concurrent spaces aren't removed

**Comment by MrMino** [comment]
Thank you! Yes, this is a confusing one. Not least because of the "indentation schism" between `mdformat` and `mkdocs`. I had situations where the hooks just briefly all passed, just to fail on the next run. I have no idea why that happens, but it does, sometimes; for repro, one should always run pre-commit twice.

**Comment by afuetterer** [comment]
I can confirm this, happens to me as well. I tested with only `mdformat` pre-commit hook and was confused, when I ran all hooks, that there are failing hooks.

**Comment by KyleKing** [comment]
Should be fixed in v2.0.7 (https://github.com/KyleKing/mdformat-mkdocs/commit/01a6916f41fb82d3e0f840d79882acb0181563af)!

**Comment by afuetterer** [comment]
Works for me, thanks a lot.
